### PR TITLE
Move out Net::HTTP adapter

### PR DESCRIPTION
**Pull request by iMacTia**
## Description
Removes Net::HTTP adapter from Faraday codebase and adds it as a dependency instead.
Replaces #1195 

## Additional Notes
This change should be completely backwards-compatible.
The dependency should be removed in Faraday v2.0

### CODE REVIEW

**Comment by iMacTia** [comment]
Just waiting for https://github.com/lostisland/faraday-net_http/pull/2 to be merged to sync the 2 repos up again, then bump the `faraday-net_http` version in the gemspec and merge this in 🎉 !

File: faraday.gemspec [hunk 1]
```diff
@@ -15,6 +15,7 @@ Gem::Specification.new do |spec|
 
   spec.required_ruby_version = '>= 2.4'
 
+  spec.add_dependency 'faraday-net_http', '~> 0.0.6'
```
Comment (olleolleolle) [inline faraday.gemspec:18]: Hm, perhaps we should start it at 1.0?

Comment (olleolleolle) [reply faraday.gemspec:18]: Like, the code is what we use in this repo.

**Review by olleolleolle — APPROVED** [review]
I love that the hackathon at Rossconf had this cool outcome!

File: faraday.gemspec (hunk 1 above)
Comment (iMacTia) [reply faraday.gemspec:18]: Yes I have already bumped `faraday-net_http` version to 1.0, BUT I realised @JanDintel is the only owner of the gem in [Rubygems](https://rubygems.org/gems/faraday-net_http) at the moment! So I can't publish it right now.
@JanDintel could you please add me and @olleolleolle as owners as well so we can publish new versions?

Comment (JanDintel) [reply faraday.gemspec:18]: @iMacTia and @olleolleolle I've added you as owners to the gem! You should have full permission to publish a new version.

@iMacTia I noticed that you still need to set-up MFA on Rubygems though :)

Comment (iMacTia) [reply faraday.gemspec:18]: @JanDintel thank you! Will release it now then, thanks a lot for all the work 🎉 

> @iMacTia I noticed that you still need to set-up MFA on Rubygems though :)

Yes, that's a looooooong story: https://github.com/rubygems/rubygems.org/issues/2500

**Comment by iMacTia** [comment]
It's done! Massive thanks to @JanDintel for doing the bulk of the work 🎉

**Comment by rmm5t** [comment]
This causes a circular dependency issue.

`faraday` requires `faraday/net_http`. [`faraday/net_http` requires faraday](https://github.com/lostisland/faraday-net_http/blob/main/lib/faraday/net_http.rb#L3). wash. rinse. repeat.

```
/path/vendor/bundle/ruby/2.5.0/gems/faraday-net_http-1.0.0/lib/faraday/net_http.rb:3: warning: loading in progress, circular require considered harmful - /Users/mcgear/work/account-service/vendor/bundle/ruby/2.5.0/gems/faraday-1.3.0/lib/faraday.rb
                from /path/vendor/bundle/ruby/2.5.0/gems/faraday-1.3.0/lib/faraday.rb:30:in  `<top (required)>'
                from /path/vendor/bundle/ruby/2.5.0/gems/faraday-1.3.0/lib/faraday.rb:30:in  `require'
                from /path/vendor/bundle/ruby/2.5.0/gems/faraday-net_http-1.0.0/lib/faraday/net_http.rb:3:in  `<top (required)>'
```

**Pull request merged**